### PR TITLE
Autobuild: Upgrade Windows build environment

### PR DESCRIPTION
**Pull request by ann0see**
Visual Studio and CodeQl now work on windows-2022 according to https://github.com/github/codeql-action/issues/850

**Short description of changes**

CHANGELOG: Autobuild: Build Windows version on latest build environment. This upgrades the Windows build process

**Context: Fixes an issue?**

Fixes: https://github.com/jamulussoftware/jamulus/issues/2259

**Does this change need documentation? What needs to be documented and how?**

No.

**Status of this Pull Request**

Needs testing on Windows. Hopefully the upgrade has only positive side-effects.

**What is missing until this pull request can be merged?**

## Checklist

-  [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
-  [x] I tested my code and it does what I want (ASIO build installs and runs)
-  [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#source-code-consistency) 
-  [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
-  [x] I've filled all the content above

### CODE REVIEW

File: .github/workflows/autobuild.yml [hunk 1]
```diff
@@ -159,13 +159,13 @@ jobs:
 
           - config_name:            Windows (artifact+codeQL)
             target_os:              windows
-            building_on_os:         windows-2019
+            building_on_os:         windows-latest
```
Comment (pljones) [inline .github/workflows/autobuild.yml:162]: Is there a pinnable version?

**Comment by pljones** [comment]
Have you run the auto-build on this branch on Github?  Are the artifacts available?

**Comment by ann0see** [comment]
Yes. You can test the artifices. According to the log they‘ve been built on the latest windows version with VS 22

File: .github/workflows/autobuild.yml (hunk 1 above)
Comment (ann0see) [reply .github/workflows/autobuild.yml:162]: Yes. windows-2022 probably

**Comment by pljones** [comment]
Well the ASIO(tm) build lets me listen okay.  I've not had a chance to test anything more than that but I can't think it would have a problem if it works that far.  I can't test JACK (it still crashes for me - JACK, not Jamulus).

File: .github/workflows/autobuild.yml (hunk 1 above)
Comment (pljones) [reply .github/workflows/autobuild.yml:162]: I'd prefer that rather than something that could change.

**Comment by ann0see** [comment]
@henkdegroot can test JACK on Windows, I suppose. But I doubt there's anything wrong with it

**Review by hoffie — APPROVED** [review]
Great, thanks!
As @pljones mentioned, please try the pinned OS version. Besides that, approving (I can't test on Windows, but I guess it's fine if @pljones and @henkdegroot do).

**Comment by henkdegroot** [comment]
@ann0see / @hoffie, annoying fact (again) I can't download the artifact. After the download completes, Microsoft detects a trojan in the jack exe. Can you try to download and see if you get the same result?

https://www.microsoft.com/en-us/wdsi/threats/malware-encyclopedia-description?name=Trojan%3aScript%2fWacatac.B!ml&threatid=2147735503

**Comment by pljones** [comment]
@henkdegroot the Windows JACK build opens fine for me -- you mean the installer, I take it?  Or do you mean _after_ install?

**Comment by henkdegroot** [comment]
@pljones I mean that when I download the jack zip file from the autobuild, Windows detects a threat and blocks the download after it completed.

**Comment by pljones** [comment]
I don't get that at all - but then, I've optimised my PC for real-time audio heavily by turning off background stuff I don't want.

**Review by pljones — APPROVED** [review]

**Comment by hoffie** [comment]
> @ann0see / @hoffie, annoying fact (again) I can't download the artifact. After the download completes, Microsoft detects a trojan in the jack exe. Can you try to download and see if you get the same result?

Is that more prevalent after this PR? I think it is expected that non-famous Jamulus builds cause warnings. However, specific detections sound like something to investigate. Interestingly, on Virustotal the files are not flagged by Microsoft (but by 1-2 other scanners in rather generic ways). Jack and ASIO seem similar though:

- Jack [ZIP](https://www.virustotal.com/gui/file/f7c9e9f600c1d9306c5c0041b8ec3db663488ec4d8d4b182feb75a53a7bb6a76/relations), [Installer](https://www.virustotal.com/gui/file/0380e9f106d42db1dd93e61e7a5b58b0fb3110ac7a3fb59bce17ecf9d32a046d?nocache=1), [Jamulus.exe](https://www.virustotal.com/gui/file/e5723dff89db9ca9f68a09116a2cf77b353b42b59e6432eb4abbc2776135214d?nocache=1)
- ASIO [ZIP](https://www.virustotal.com/gui/file/313e89421e8a28edff52548d4632787ee4e0c0d4d69bab6a9f53626601f62638?nocache=1) (haven't checked the extracted parts)

**Review by hoffie — APPROVED** [review]
I don't see a VirusTotal regression when comparing the artifacts from here with one of the latest master builds:
https://www.virustotal.com/gui/file-analysis/MmQ1YjQ4N2RlMjRkODY3YjZlOGU2ZTc2OTkyZGMzYjM6MTY1NTU4NDkyMQ==

Therefore, approving.

**Comment by henkdegroot** [comment]
@hoffie, no problems with warnings. I did not have any problem downloading the ASIO version, only the windows jack version. Interesting to see why Microsoft Defender is triggering this high security and blocks the file. When I unblock the zip file and try to extract it I get more (other) trojan warnings being detected. Just annoying....

**Comment by henkdegroot** [comment]
FWIW, I actually got feedback from Microsoft about the file being false positive and they have updated their signatures....will see if these submissions can really help for future cases as well.

**Pull request merged**